.travis.yml

```diff
@@ -0,0 +1,36 @@
+sudo: required
+language: rust
+rust:
+  - nightly
+  - beta
+  - stable
+
+env:
+  - GTK=3.4
+  - GTK=3.18
+
+addons:
+  apt:
+    packages:
+      - libcurl4-openssl-dev
+      - libelf-dev
+      - libdw-dev
+      - binutils-dev
+      - libgtk-3-dev
+      - libsoup2.4-dev
+      - libxtst-dev
+      - at-spi2-core
+      - openbox
+
+before_script:
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+  - sleep 3 # give xvfb some time to start
+  - openbox & # needs a window manager for the tests (to get focus to work)
+
+script:
+  - rustc --version
+  - cargo install clippy || touch clippy_failed
+  - if [ ! -f clippy_failed ]; then cargo clippy; fi
```

Comment:
Remove clippy for now, we can add that later. For now it would only fail the build as there are warnings left... and we can only install clippy anyway if using nightly (so you need to check if we're running the nightly configuration). You can check that via `"$TRAVIS_RUST_VERSION" = "nightly"`